development/transitions.md

```diff
@@ -42,16 +43,14 @@ Transitions obey the following rules:
 
 Callback arguments:
 
-* callback(err, true)
-
-  The document is saved and `ok` property value is false if `err` is truthy.
-
-* callback(err)
+* callback(err, needsSaving)
 
-  The document is saved and `ok` property value is false if `err` is truthy.
-  Use this when a transition fails and should be re-run.
+   `needsSaving` is true if the `change.doc` needs to be saved to db by the transition runner. For instance the transition has edited the `change.doc` in memory.
+   `err` if truthy, the error will be added to the `changes.doc` in memory. (Note that if `needsSaving` is falsy, the doc will not be saved, so that error will not be persisted).
 
-* callback()
+Regardless whether the doc is saved or not, the transitions will all be run (unless one crashes!).
 
-  Nothing to be done, the document is not saved and next transition continues.
-  Transitions will run again on next change.
+When your transition encounters an error, there are different ways to deal with it. You can :
+- finish your transition with `callback(someError, true)`. This will save the error to `change.doc`. The transition will not be rerun on this doc.
```

Comment:
Wow is that true? That due to an error we effectively blacklist documents for ever? That seems like a dangerous automatic process to employ